library/source/open_cdm.cpp

```diff
@@ -411,3 +411,19 @@ OpenCDMError opencdm_session_decrypt(struct OpenCDMSession *session, uint8_t enc
     kLog << warn << __func__ << " not implemented";
     return ERROR_FAIL;
 }
+
+OpenCDMError opencdm_get_metric_system_data(struct OpenCDMSystem* system, uint32_t* bufferLength, std::vector<uint8_t> &buffer)
```

Comment:
You shouldn't change the ocdm's API. It should remain the same as in ocdm, ie. the buffer should have uint8_t* type.
Vector should be used in Rialto's APIs.